[JBIDE-26165] - Implement a Launcher project wizard

jeffmaury opened the pull request:
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardPage.java:
```diff
@@ -0,0 +1,182 @@
+			WizardUtils.runInWizard(Job.create("Loading launcher catalog", monitor -> model.setCatalog(CatalogManager.getDefault().getCatalog(monitor))), getContainer());
+		} catch (InvocationTargetException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
* ignore or log? 
* unify in catch block with InterruptedException?

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardPage.java:
```diff
@@ -0,0 +1,182 @@
+				.to(BeanProperties.value(NewLauncherProjectModel.USE_DEFAULT_LOCATION_PROPERTY).observe(model)).in(dbc);
+
+		// location
+		//Widget txtLocation = createTextWidget(parent, dbc, "Location", NewLauncherProjectModel.LOCATION_PROPERTY, "Please specify a location for your project");
```
remove commented line?

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardPage.java:
```diff
@@ -0,0 +1,182 @@
+		buttonUseDefaultLocation.setText("Use default location");
+		GridDataFactory.fillDefaults().align(SWT.LEFT, SWT.CENTER).span(2, 1).applyTo(buttonUseDefaultLocation);
+		IObservableValue useDefaultLocationButtonObservable = WidgetProperties.selection().observe(buttonUseDefaultLocation);
+		Binding useDefaultLocationBinding = ValueBindingBuilder.bind(useDefaultLocationButtonObservable)
```
useless assignment to local variable

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardPage.java:
```diff
@@ -0,0 +1,182 @@
+		GridLayoutFactory.fillDefaults().margins(6, 6).numColumns(2).applyTo(parent);
+
+		// project name
+		createTextWidget(parent, dbc, "Project name", NewLauncherProjectModel.PROJECT_NAME_PROPERTY, new EclipseProjectValidator("Please specify an Eclipse project", "Project already exists"));
```
extensively long line, 191 characters wide.

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardPage.java:
```diff
@@ -0,0 +1,182 @@
+				.converting(IConverter.create(String.class, IPath.class, NewLauncherProjectWizardPage::string2IPath))
+				.to(BeanProperties.value(NewLauncherProjectModel.LOCATION_PROPERTY).observe(model)).in(dbc);
+		IObservableValue locationTextEnabledObservable = WidgetProperties.enabled().observe(txtLocation);
+		Binding locationEnabledBinding = ValueBindingBuilder.bind(locationTextEnabledObservable)
```
useless assignment to local variable

adietish commented on common/plugins/org.jboss.tools.common.launcher.core/src/main/java/org/jboss/tools/common/launcher/core/model/CatalogManager.java:
```diff
@@ -12,17 +12,28 @@
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
+import org.jboss.tools.common.launcher.core.LauncherCoreConstants;
```
import never used

adietish commented on common/plugins/org.jboss.tools.common.launcher.core/src/main/java/org/jboss/tools/common/launcher/core/model/CatalogManager.java:
```diff
@@ -70,4 +81,25 @@ private Catalog loadCatalog(String endpointURL, IProgressMonitor monitor) throws
+		endpointURL += LAUNCHER_ZIP_PROJECT_NAME_PARAMETER_NAME + '=' + projectName + '&';
+		endpointURL += LAUNCHER_ZIP_GROUP_ID_PARAMETER_NAME + '=' + groupId + '&';
+		endpointURL += LAUNCHER_ZIP_ARTIFACT_ID_PARAMETER_NAME + '=' + artifactId + '&';
+		endpointURL += LAUNCHER_ZIP_VERSION_PARAMETER_NAME + '=' + version;
```
having a test for CatalogManager would be great

adietish commented on common/plugins/org.jboss.tools.common.launcher.core/src/main/java/org/jboss/tools/common/launcher/core/model/Mission.java:
```diff
@@ -47,10 +47,37 @@ public void setName(String name) {
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```
simple test for this method would be great

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectModel.java:
```diff
@@ -0,0 +1,233 @@
+import org.jboss.tools.common.launcher.core.model.Catalog;
+import org.jboss.tools.common.launcher.core.model.Mission;
+
+public class NewLauncherProjectModel extends ObservablePojo {
```
a test for NewLauncherProjectModel would be great

adietish commented on common/plugins/org.jboss.tools.common.launcher.ui/src/main/java/org/jboss/tools/common/launcher/ui/wizard/NewLauncherProjectWizardController.java:
```diff
@@ -0,0 +1,197 @@
+			if (!monitor.isCanceled() && status.isOK()) {
+				status = unzip(output.toByteArray(), model.getLocation());
+			}
+		} finally {
```
I would think that there's no need for this finally block

jeffmaury commented:
My remarks:

- I choose base because it seems to me the logical place to be and m2e dependency is not an issue because it's not the default in Eclipse distributions

- Updated page description and added tooltips for mission and runtime
- the error is raised by data binding and the RequiredControlDecorationUpdater
- the wizard is not Maven specific (think of NodeJS) and the M2E Maven wizard does not validate the version string
- Regarding the groovy JAR I also add issues with dependencies because it seems my M2 repo was garbage due to Nexus down on Aug 4th. Once I cleaned the failed dependencies it was ok for me

adietish commented:
@jeffmaury 
> the error is raised by data binding and the RequiredControlDecorationUpdater

Looks like you're using the decorator that supports the "*" marker. This seems to indicate that you'd get the "*" decoration if you validate to CANCEL (instead of ERROR). Worth changing?

jeffmaury commented:
The ERROR is generated by org.jboss.tools.common.ui.databinding.MandatoryStringValidator which is a utility class. Do you think we can switch to CANCEL ?

adietish commented:
we agreed that I'll add the suggested improvements in the next sprint. Things work for me. +1
